crates/sui-indexer-alt-framework/src/handlers/cp_sequence_numbers.rs

```diff
@@ -22,6 +23,80 @@ pub struct StoredCpSequenceNumbers {
 
 pub struct CpSequenceNumbers;
 
+/// A struct that can be instantiated by the pruner task to map a `from` and `to` checkpoint to its
+/// corresponding `tx_lo` and containing epoch. The `from` checkpoint is expected to be inclusive,
+/// and the `to` checkpoint is exclusive. This requires the existence of the `checkpoint_metadata`
+/// table.
+pub struct PrunableRange {
+    from: StoredCpSequenceNumbers,
+    to: StoredCpSequenceNumbers,
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum RangeError {
+    #[error("Invalid checkpoint range: `from` {0} is greater than `to` {1}")]
+    InvalidCheckpointRange(u64, u64),
+    #[error("No checkpoint mapping found for checkpoints {0} and {1}")]
+    NoCheckpointMapping(u64, u64),
+    #[error("Database error: {0}")]
+    DbError(diesel::result::Error),
+}
```

Comment:
I think using an `anyhow::Error` would be fine FWIW -- I would only create a new structured error type if I wanted some more flexibility in doing conversions between errors, or if I expected people to care about what kind of error they got and programmatically change behaviour based on that. It seems unlikely that someone would do that for this error.